### PR TITLE
data processing + processing for futureSuitability 

### PR DESCRIPTION
**Pull request by lindsayplatt**
addressing #38 and some of #19

### CODE REVIEW

**Comment by jordansread** [comment]
@lindsaycarr can you fix conflicts?

**Comment by lindsayplatt** [comment]
@aappling-usgs can you review my new changes? Still unsure about what the first argument to `visualize.visualizeFishTrends` will be since visualizeBassTrends and visualizeWallyTrends both call it

File: scripts/visualize/visualizeWalleyeDecline.R [hunk 1]
```diff
@@ -20,7 +20,7 @@ svgFishTrends <- function(object, filename){
   dinosvg:::write_svg(svg, file = filename)
 }
 
-visualize.visualizeFishTrends <- function(item){
+visualize.visualizeFishTrends <- function(outfile, ...){
   fish <- readData(item$filename)
```
Comment (aappling-usgs) [inline scripts/visualize/visualizeWalleyeDecline.R]: this function can+should expect arguments named for the IDs specified in the `depends` field for the figure items in viz.yaml. that's how you can tell them apart. (one call will include `processedWallyTrends=readData('processedWallyTrends')` as an argument; the other will include `processedBassTrends=readData('processedBassTrends')`

File: viz.yaml [hunk 2]
```diff
@@ -80,71 +80,108 @@ fetch:
...
-    fig.config:
+    args:
         col: '#01b29F'
   -
     id: bassTrendsFig
     location: figures/bassTrendsFig.svg
-    depends: bassTrends
+    depends: processedBassTrends
+    scripts: scripts/visualize/visualizeWalleyeDecline.R
     visualizer: visualizeFishTrends
```
Comment (aappling-usgs) [inline viz.yaml:166]: yep, exactly

File: scripts/visualize/visualizeWalleyeDecline.R (hunk 1 above)
Comment (aappling-usgs) [inline scripts/visualize/visualizeWalleyeDecline.R]: i.e.:

``` r
visualize.visualizeFishTrends <- function(processedWallyTrends, processedBassTrends, outfile, ...) {
  if(!missing(processedWallyTrends)) {
    fish <- processedWallyTrends
    config <- list(col='red')
  } else {
    fish <- processedBassTrends
    config <- list(col='blue')
  }
```

Comment (lindsayplatt) [inline scripts/visualize/visualizeWalleyeDecline.R]: :+1: and then I get rid of the 

```
args: 
     col: '#01b29F'
```

in viz.yaml, correct?

Comment (aappling-usgs) [inline scripts/visualize/visualizeWalleyeDecline.R]: maybe - it gets ignored for now. i would like to implement args soonish, so you could keep it in as a placeholder if you want...

Comment (lindsayplatt) [inline scripts/visualize/visualizeWalleyeDecline.R]: whoops - I deleted before seeing this, so we will have to put it back in

Comment (aappling-usgs) [inline scripts/visualize/visualizeWalleyeDecline.R]: no worries either way. it doesn't do anything right now.

**Pull request merged**